acme/api/handler.go

```diff
@@ -184,10 +215,25 @@ func (h *Handler) GetChallenge(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	getLink := h.Auth.GetLink
-	w.Header().Add("Link", link(getLink(acme.AuthzLink, acme.URLSafeProvisionerName(prov), true, ch.GetAuthzID()), "up"))
-	w.Header().Set("Location", getLink(acme.ChallengeLink, acme.URLSafeProvisionerName(prov), true, ch.GetID()))
-	api.JSON(w, ch)
+	switch ch.Status {
+	case acme.StatusPending:
+		panic("validation attempt did not move challenge to the processing state")
```

Comment:
panic? If it's an internal server error then let's return that. I don't think we want to panic.